### PR TITLE
Restructure import/export section and add grid layout and community package migrator documentation 

### PR DESCRIPTION
**Pull request by AndyButland**
## Description

Restructured import/export section and add grid layout and community package migrator documentation.

This PR extends on and supersedes https://github.com/umbraco/UmbracoDocs/pull/6284 which can be closed.

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [ ] New content
* [X] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

Deploy 13.2.0-rc1 and 14.1.0-rc1
Deploy Contrib 4.4.0-rc1, 10.3.0-rc1, 12.2.0-rc1, 13.2.0-rc1 and 14.1.0-rc1

## Deadline (if relevant)

Can be reviewed and published at any time.

### CODE REVIEW

**Comment by eshanrnh** [comment]
Thanks, @AndyButland for the PR 🙌 We will review it as soon as we can 💪

File: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+- `IArtifactJsonMigrator` - where the migration occurs at the lower level of transforming the serialized JSON itself.
+
+Implementations to handle common migrations of Data Types from obsoleted property editors are available:
+
+- `ReplaceMediaPickerDataTypeArtifactMigrator` - migrates a Data Type from using the legacy Media Picker to the current version of this property editor
+- `ReplaceNestedContentDataTypeArtifactMigrator` - migrates a Data Type based on the obsolete Nested Content property editor to the Block List
+- `ReplaceGridDataTypeArtifactMigrator` - migrates a Data Type based on the legacy Grid layout into the Block Grid
+- `ReplaceUnknownEditorDataTypeArtifactMigrator` - replaces any unknown editor alias with a label
+
+We've also made available base implementations that you can use to build your own migrations. You may have a need to handle transfer of information between other obsolete and replacement property editors that you have in your Umbraco application.
```
Comment (eshanrnh) [inline 13/umbraco-deploy/deployment-workflow/import-with-migrations.md:32]:
```suggestion
We've also made available base implementations that you can use to build your own migrations. You may need to handle the transfer of information between other obsolete and replacement property editors that you have in your Umbraco application.
```

File: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+- `PropertyTypeMigratorBase`
+- `GridPropertyTypeMigratorBase`
+
+{% hint style="info" %}
+Property editor changes are determined by comparing the `PropertyEditorAliases` dictionary stored in the content artifact to the current Content Type/Data Type configuration. The dictionary contains editor aliases for each content property.
+{% endhint %}
+
+### Registering migrators
+
+Migrators will run if you've registered them to, hence you can enable only the ones needed for your solution.
```
Comment (eshanrnh) [inline 13/umbraco-deploy/deployment-workflow/import-with-migrations.md:65]:
```suggestion
Migrators will run if you have registered them, so you can enable only the ones needed for your solution.
```

File: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+            .Append<MediaPickerPropertyTypeMigrator>();
+    }
+}
+ ```
+
+### Details of Specific Migrations
+
+Umbraco Deploy ships with migrators to handle the conversion of core property editors as they have changed, been removed or replaced between versions.
+
+Open source migrators may be built by HQ or the community for property editors found in community packages. They will be made them available for [use](https://www.nuget.org/packages/Umbraco.Deploy.Contrib) and [review](https://github.com/umbraco/Umbraco.Deploy.Contrib/tree/v13/dev/src/Umbraco.Deploy.Contrib/Migrators) via the `Umbraco.Deploy.Contrib` package.
```
Comment (eshanrnh) [inline 13/umbraco-deploy/deployment-workflow/import-with-migrations.md:93]:
```suggestion
Open source migrators may be built by HQ or the community for property editors found in community packages. They will be made available for [use](https://www.nuget.org/packages/Umbraco.Deploy.Contrib) and [review](https://github.com/umbraco/Umbraco.Deploy.Contrib/tree/v13/dev/src/Umbraco.Deploy.Contrib/Migrators) via the `Umbraco.Deploy.Contrib` package.
```

File: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+        builder.DeployArtifactMigrators()
+            .Append<ReplaceGridDataTypeArtifactMigrator>();
+
+        builder.DeployPropertyTypeMigrators()
+            .Append<GridPropertyTypeMigrator>();
+    }
+}
+```
+
+Thee implementations make use of the following conventions to migrate the data:
```
Comment (eshanrnh) [inline 13/umbraco-deploy/deployment-workflow/import-with-migrations.md:120]:
```suggestion
These implementations make use of the following conventions to migrate the data:
```

File: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+      - `treepicker`, `treesource`, `textarea` or any other - the default 'Textarea', falling back to the first `Umbraco.TextArea` editor.
+- `GridPropertyTypeMigrator`:
+  - Gets the grid layout and row configuration element types based on the alias prefix/name convention used by the Data Type artifact migrator;
+  - The grid editor values are migrated to the respective properties:
+    - The `media` grid editor converts the value to a media item with crops (based on the UDI or media path), including the focal point (although this needs to be enabled on the Data Type), alternate text and caption;
+    - All other values are converted to a text value or otherwise to a JSON string;
+  - If a row or cell contains settings config or styles and the corresponding block has a settings element type configured, the settings config and styles are migrated to their respective properties in a similar way, based on the property editor alias:
+    - `Umbraco.MediaPicker3` - removes `url('` from the beginning and `')` from the end of the value (commonly used as modifier and added to the stored value), before trying to get the media item by path.
```
Comment (eshanrnh) [inline 13/umbraco-deploy/deployment-workflow/import-with-migrations.md:150]:
```suggestion
    - `Umbraco.MediaPicker3` - removes `url('` from the beginning and `')` from the end of the value (commonly used as a modifier and added to the stored value), before trying to get the media item by a path.
```

File: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+    - All other values are returned as-is.
+
+Given the flexibility of the grid editor and Block Grid you may want to take further control over the migration. You can do that by creating your own migrator classes, that make use of our provided base classes. You would then register your own migrators instead of the ones shipped with Umbraco Deploy in your composer.
+
+The base classes provide the following functionality. Methods you should look to override to amend the default behavior have been noted above.
+
+- `ReplaceGridDataTypeArtifactMigratorBase` - replaces the `Umbraco.Grid` Data Type editor alias with `Umbraco.BlockGrid` and migrates the configuration:
+  - The amount of columns is copied over.
```
Comment (eshanrnh) [inline 13/umbraco-deploy/deployment-workflow/import-with-migrations.md:158]:
```suggestion
  - The number of columns is copied over.
```

File: 13/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+            .Append<ReplaceMatryoshkaArtifactMigrator>();
+    }
+}
+```
+
+### Source Code Example -  Nested Content to Block List
+
+As described above, the nested content to block list migration will occur register the corresponding migrator with your application.
+
+In order to help writing your own migrations. we share here the source code of an example that ships with Umbraco Deploy. This migration converts Nested Content to Block List.
```
Comment (eshanrnh) [inline 13/umbraco-deploy/deployment-workflow/import-with-migrations.md:231]:
```suggestion
To help write your own migrations, we share the source code of an example that ships with Umbraco Deploy. This migration converts Nested Content to Block List.
```

File: 14/umbraco-deploy/deployment-workflow/import-export-v7.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
+---
+meta.Title: Migrating from Umbraco 7
+description: How to export content and schema from Umbraco 7 and import into a newer version
+---
+
+### Migrating from Umbraco 7
+
+The import and export feature is available for Umbraco Deploy supporting Umbraco 8 and above. It's not been ported back to Umbraco 7, hence you can't trigger an export from there in the same way.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-export-v7.md:8]:
```suggestion
The import and export features are available for Umbraco Deploy supporting Umbraco 8 and above. It's not been ported back to Umbraco 7, hence you cannot trigger an export from there in the same way.
```

File: 14/umbraco-deploy/deployment-workflow/import-export-v7.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+
+We can however use this feature to help migrate from Umbraco 7 to a supported major version. It requires additional logic added to the Deploy Contrib project.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-export-v7.md:10]:
```suggestion
We can use this feature to help migrate from Umbraco 7 to a supported major version. However, it requires additional logic to be added to the Deploy Contrib project.
```

File: 14/umbraco-deploy/deployment-workflow/import-export-v7.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+
+</details>
+
+#### Importing Umbraco 7 content and schema
+
+To import this archive into a newer Umbraco project, you need to install either of these packages:
+
+- `UmbracoDeploy.Contrib` 4.3 for Umbraco 8
+- `Umbraco.Deploy.Contrib` for Umbraco 10.2, 12.1, 13.1 or later
+Then you need to configure the legacy artifact type resolver and migratory.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-export-v7.md:97]:
```suggestion
- `Umbraco.Deploy.Contrib` for Umbraco 10.2, 12.1, 13.1 or later

Then you need to configure the legacy artifact type resolver and migratory.
```

File: 14/umbraco-deploy/deployment-workflow/import-export-v7.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+
+Artifact type resolvers allow resolving changes in the type that's stored in the `__type` JSON property of the artifact. This is in case it moved to a different assembly or namespace (or got renamed) in a newer version. The legacy migrators handle the following changes:
+
+- Moving the pre-values of data types to the configuration property;
+- Moving the invariant release and expire dates of content to the (culture variant) schedule property;
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-export-v7.md:102]:
```suggestion
- Moving the invariant release and expiration dates of content to the (culture variant) schedule property;
```

File: 14/umbraco-deploy/deployment-workflow/import-export-v7.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+  - `Umbraco.RelatedLinks` to `Umbraco.MultiUrlPicker`
+  - `Umbraco.Textbox` to `Umbraco.TextBox`
+  - `Umbraco.TextboxMultiple` to `Umbraco.TextArea`
+  - `Umbraco.TinyMCEv3` to `Umbraco.TinyMCE`
+- Migrating pre-value property values for:
+  - `Umbraco.CheckBoxList`
+  - `Umbraco.DropDown.Flexible`
+  - `Umbraco.RadioButtonList`
+
+The following composer adds the required legacy artifact type resolver and migrators. It also adds a custom resolver that marks the specified Document Type alias `testElement` as element type. Element types are a concept added in Umbraco 8 and are required for document types that are used in Nested Content.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-export-v7.md:133]:
```suggestion
The following composer adds the required legacy artifact type resolver and migrators. It also adds a custom resolver that marks the specified Document Type alias `testElement` as the element type. Element types are a concept added in Umbraco 8 and are required for Document Types that are used in Nested Content.
```

File: 14/umbraco-deploy/deployment-workflow/import-export-v7.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+            : base("testElement")
+        { }
+    }
+}
+```
+
+</details>
+
+{% hint style="info" %}
+It is recommended to first only import schema and schema files (by deselecting 'Content' and 'Content files' in the dialog). You can follow that by a complete import of all content and schema. The order in which the artifacts are imported depends on the dependencies between them. By importing schema first we ensure the schema is updated before any content is processed.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-export-v7.md:166]:
```suggestion
It is recommended to start by importing only the schema and schema files (by deselecting 'Content' and 'Content files' in the dialog). Then, you can proceed with importing all content and schema together. The order in which the artifacts are imported depends on their dependencies. By importing the schema first, we ensure that the schema is updated before any content is processed.
```

File: 14/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 14]
```diff
@@ -0,0 +1,428 @@
...
+- `IArtifactJsonMigrator` - where the migration occurs at the lower level of transforming the serialized JSON itself.
+
+Implementations to handle common migrations of Data Types from obsoleted property editors are available:
+
+- `ReplaceMediaPickerDataTypeArtifactMigrator` - migrates a Data Type from using the legacy Media Picker to the current version of this property editor
+- `ReplaceNestedContentDataTypeArtifactMigrator` - migrates a Data Type based on the obsolete Nested Content property editor to the Block List
+- `ReplaceGridDataTypeArtifactMigrator` - migrates a Data Type based on the legacy Grid layout into the Block Grid
+- `ReplaceUnknownEditorDataTypeArtifactMigrator` - replaces any unknown editor alias with a label
+
+We've also made available base implementations that you can use to build your own migrations. You may have a need to handle transfer of information between other obsolete and replacement property editors that you have in your Umbraco application.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-with-migrations.md:32]:
```suggestion
We've also made available base implementations that you can use to build your own migrations. You may need to handle the transfer of information between other obsolete and replacement property editors that you have in your Umbraco application.
```

File: 14/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 15]
```diff
@@ -0,0 +1,428 @@
...
+- `PropertyTypeMigratorBase`
+- `GridPropertyTypeMigratorBase`
+
+{% hint style="info" %}
+Property editor changes are determined by comparing the `PropertyEditorAliases` dictionary stored in the content artifact to the current Content Type/Data Type configuration. The dictionary contains editor aliases for each content property.
+{% endhint %}
+
+### Registering migrators
+
+Migrators will run if you've registered them to, hence you can enable only the ones needed for your solution.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-with-migrations.md:65]:
```suggestion
Migrators will run if you have registered them, so you can enable only the ones needed for your solution.
```

File: 14/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 16]
```diff
@@ -0,0 +1,428 @@
...
+            .Append<MediaPickerPropertyTypeMigrator>();
+    }
+}
+ ```
+
+### Details of Specific Migrations
+
+Umbraco Deploy ships with migrators to handle the conversion of core property editors as they have changed, been removed or replaced between versions.
+
+Open source migrators may be built by HQ or the community for property editors found in community packages. They will be made them available for [use](https://www.nuget.org/packages/Umbraco.Deploy.Contrib) and [review](https://github.com/umbraco/Umbraco.Deploy.Contrib/tree/v13/dev/src/Umbraco.Deploy.Contrib/Migrators) via the `Umbraco.Deploy.Contrib` package.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-with-migrations.md:93]:
```suggestion
Open source migrators may be built by HQ or the community for property editors found in community packages. They will be made available for [use](https://www.nuget.org/packages/Umbraco.Deploy.Contrib) and [review](https://github.com/umbraco/Umbraco.Deploy.Contrib/tree/v13/dev/src/Umbraco.Deploy.Contrib/Migrators) via the `Umbraco.Deploy.Contrib` package.
```

File: 14/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 17]
```diff
@@ -0,0 +1,428 @@
...
+        builder.DeployArtifactMigrators()
+            .Append<ReplaceGridDataTypeArtifactMigrator>();
+
+        builder.DeployPropertyTypeMigrators()
+            .Append<GridPropertyTypeMigrator>();
+    }
+}
+```
+
+Thee implementations make use of the following conventions to migrate the data:
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-with-migrations.md:120]:
```suggestion
These implementations make use of the following conventions to migrate the data:
```

File: 14/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 18]
```diff
@@ -0,0 +1,428 @@
...
+      - `treepicker`, `treesource`, `textarea` or any other - the default 'Textarea', falling back to the first `Umbraco.TextArea` editor.
+- `GridPropertyTypeMigrator`:
+  - Gets the grid layout and row configuration element types based on the alias prefix/name convention used by the Data Type artifact migrator;
+  - The grid editor values are migrated to the respective properties:
+    - The `media` grid editor converts the value to a media item with crops (based on the UDI or media path), including the focal point (although this needs to be enabled on the Data Type), alternate text and caption;
+    - All other values are converted to a text value or otherwise to a JSON string;
+  - If a row or cell contains settings config or styles and the corresponding block has a settings element type configured, the settings config and styles are migrated to their respective properties in a similar way, based on the property editor alias:
+    - `Umbraco.MediaPicker3` - removes `url('` from the beginning and `')` from the end of the value (commonly used as modifier and added to the stored value), before trying to get the media item by path.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-with-migrations.md:150]:
```suggestion
    - `Umbraco.MediaPicker3` - removes `url('` from the beginning and `')` from the end of the value (commonly used as a modifier and added to the stored value), before trying to get the media item by a path.
```

File: 14/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 19]
```diff
@@ -0,0 +1,428 @@
...
+    - All other values are returned as-is.
+
+Given the flexibility of the grid editor and Block Grid you may want to take further control over the migration. You can do that by creating your own migrator classes, that make use of our provided base classes. You would then register your own migrators instead of the ones shipped with Umbraco Deploy in your composer.
+
+The base classes provide the following functionality. Methods you should look to override to amend the default behavior have been noted above.
+
+- `ReplaceGridDataTypeArtifactMigratorBase` - replaces the `Umbraco.Grid` Data Type editor alias with `Umbraco.BlockGrid` and migrates the configuration:
+  - The amount of columns is copied over.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-with-migrations.md:158]:
```suggestion
  - The number of columns is copied over.
```

File: 14/umbraco-deploy/deployment-workflow/import-with-migrations.md [hunk 20]
```diff
@@ -0,0 +1,428 @@
...
+            .Append<ReplaceMatryoshkaArtifactMigrator>();
+    }
+}
+```
+
+### Source Code Example -  Nested Content to Block List
+
+As described above, the nested content to block list migration will occur register the corresponding migrator with your application.
+
+In order to help writing your own migrations. we share here the source code of an example that ships with Umbraco Deploy. This migration converts Nested Content to Block List.
```
Comment (eshanrnh) [inline 14/umbraco-deploy/deployment-workflow/import-with-migrations.md:231]:
```suggestion
To help write your own migrations, we share the source code of an example that ships with Umbraco Deploy. This migration converts Nested Content to Block List.
```

**Review by eshanrnh — CHANGES_REQUESTED** [review]
The PR looks good, @AndyButland 🙌 I'll incorporate the suggestions. 

Can you please add the v13 Summary file changes to v14 as well

**Comment by AndyButland** [comment]
Thanks @eshanrnh - I've updated the 14 summary file.

**Review by eshanrnh — APPROVED** [review]
Looks good to me. 🙌

**Pull request merged**